gamemode/sh_items.lua

```diff
@@ -300,21 +300,26 @@ UPGRADEGROUP_T6 = 7
 
 GM.UpgradeGroups = {}
 function GM:AddUpgradeGroup(_groupkey, _points, _name, _clips)
-	local tab = {points = _points, name = _name, clips = _clips}
+  local _signature = string.lower(_name)
+  _signature = string.gsub(_signature, "%p", "")
+
+  _signature = string.gsub(_signature, "%s", "")
+
+	local tab = {points = _points, name = _name, signature = _signature, clips = _clips}
 
 	self.UpgradeGroups[_groupkey] = tab
 
 	return tab
 end
 
 --maintain ascending order by UPGRADEGROUP key or it will cause problems
-GM:AddUpgradeGroup(UPGRADEGROUP_T1, -1, "Tier 1: Starter Pistols")
-GM:AddUpgradeGroup(UPGRADEGROUP_T1_MELEE, -1, "Tier 1: Starter Melee")
-GM:AddUpgradeGroup(UPGRADEGROUP_T2, 25, "Tier 2: Pistols")
-GM:AddUpgradeGroup(UPGRADEGROUP_T3, 75, "Tier 3: SMGs")
-GM:AddUpgradeGroup(UPGRADEGROUP_T4, 125, "Tier 4: Shotguns")
-GM:AddUpgradeGroup(UPGRADEGROUP_T5, 200, "Tier 5: Assault Weapons")
-GM:AddUpgradeGroup(UPGRADEGROUP_T6, 300, "Tier 6: Pulse Weapons")
+GM:AddUpgradeGroup(UPGRADEGROUP_T1, -1, "Guns")
+GM:AddUpgradeGroup(UPGRADEGROUP_T1_MELEE, -1, "Melee")
+GM:AddUpgradeGroup(UPGRADEGROUP_T2, 25, "Pistol")
+GM:AddUpgradeGroup(UPGRADEGROUP_T3, 75, "SMG")
+GM:AddUpgradeGroup(UPGRADEGROUP_T4, 125, "Shotguns")
+GM:AddUpgradeGroup(UPGRADEGROUP_T5, 200, "Assault Weapons")
+GM:AddUpgradeGroup(UPGRADEGROUP_T6, 300, "Pulse Weapons")
```

Comment:
these names should be more clearly translated text, "tr_melee", etc.
I'd like to separate the tier numbers out into their own variable, so instead of having tr_melee holding the whole string, you'd have it like this:

translate("tr_tier") .. " " .. tiernumber .. ": " .. translate("tr_shotguns")

Reply:
k